tests/test_pseudo_points.cc

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file test_pseudo_points.cc
+ *
+ * @author Felix Musil <felix.musil@epfl.ch>
+ *
+ * @date   7 January 2020
+ *
+ * @brief test the implementation of pseudo_points classes
+ *
+ * @section LICENSE
+ *
+ * Copyright  2020 Felix Musil, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "test_adaptor.hh"
+#include "test_calculator.hh"
+#include "test_manager_collection.hh"
+
+#include "rascal/models/pseudo_points.hh"
+#include "rascal/utils/json_io.hh"
+
+#include <boost/mpl/list.hpp>
+#include <boost/test/unit_test.hpp>
+
+namespace rascal {
+  BOOST_AUTO_TEST_SUITE(pseudo_points_test);
+
+  using ManagerCollection_t =
+      ManagerCollection<StructureManagerCenters, AdaptorNeighbourList,
+                        AdaptorCenterContribution, AdaptorStrict>;
+
+  template <class Representation, class ManagerCollection,
+            template <class> class PseudoPoints>
+  struct PseudoPointsFixture {
+    using Representation_t = Representation;
+    using ManagerCollection_t = ManagerCollection;
+    using PseudoPoints_t = PseudoPoints<Representation>;
+
+    PseudoPointsFixture() {
+      hypers["cutoff_function"] = fc_hypers;
+      hypers["gaussian_density"] = sigma_hypers;
+      hypers["radial_contribution"] = {{"type", "GTO"}};
+
+      json ad1a{{"name", "AdaptorNeighbourList"},
+                {"initialization_arguments", {{"cutoff", cutoff}}}};
+      json ad1b{{"name", "AdaptorCenterContribution"},
+                {"initialization_arguments", {}}};
+      json ad2{{"name", "AdaptorStrict"},
+               {"initialization_arguments", {{"cutoff", cutoff}}}};
+      adaptors.emplace_back(ad1a);
+      adaptors.emplace_back(ad1b);
+      adaptors.emplace_back(ad2);
+    }
+
+    std::string filename{"reference_data/inputs/small_molecules-20.json"};
+
+    json adaptors{};
+
+    PseudoPoints_t sparse_points{};
+
+    double cutoff{3.};
+    json hypers{{"max_radial", 1},
+                {"max_angular", 1},
+                {"compute_gradients", true},
```

Comment:
Are gradients needed to check the pseudo points?
Gradients themselves are check somewhere else, right?
Maybe here is a possibility to have a smaller test.

Reply:
right nice catch, its en unfortunate copy/paste.